services-bytebuf-codec/src/main/java/io/scalecube/services/transport/api/DefaultHeadersCodec.java

```diff
@@ -0,0 +1,84 @@
+package io.scalecube.services.transport.api;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/** Simple binary codec for headers service message. */
+public class DefaultHeadersCodec implements HeadersCodec {
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public String contentType() {
+    return "application/octet-stream";
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public void encode(OutputStream stream, Map<String, String> headers) throws IOException {
+    if (headers.isEmpty()) {
+      return;
+    }
+    writeInt(stream, headers.size());
+    for (Entry<String, String> header : headers.entrySet()) {
+      byte[] nameBytes = header.getKey().getBytes(UTF_8);
+      byte[] valueBytes = header.getValue().getBytes(UTF_8);
+      writeInt(stream, nameBytes.length);
+      writeInt(stream, valueBytes.length);
+      for (byte nameByte : nameBytes) {
```

Comment:
Use this method `java.io.OutputStream#write(byte[])`  instead of iterating by each byte value